bentoml/_internal/bento/build_config.py

```diff
@@ -110,28 +107,29 @@ def _envars_validator(
     attribute: "Attribute[t.Union[str, t.List[str], t.Dict[str, t.Any]]]",
     value: t.Union[str, t.List[str], t.Dict[str, t.Any]],
 ) -> None:
-    if isinstance(value, str):
-        if not os.path.exists(value) and os.path.isfile(value):
-            raise InvalidArgument(
-                f'`env="{value}"` is not a valid path. if `env` is parsed as a string, it should be a path to an ".env" file.'
-            )
-
-    envars_rgx = re.compile(r"(^[A-Z0-9_]+)(\=)(.*\n)")
-
     if isinstance(value, list):
-        for envar in value:
-            match = envars_rgx.match(envar)
-            if match is None:
-                raise InvalidArgument(f"{envar} doesn't follow format ENVAR=value")
-
-            k, _ = envar.split("=")
-            if not k.isupper():
+        for envar in value:  # pragma: no cover
+            #  we have tests for this cases, but pytest refuse to cover
+            envars_rgx = re.compile(r"(^[A-Z_]+\w*)(\=)(.*)")
+            if not envars_rgx.match(envar):
                 raise InvalidArgument(
-                    f'Invalid env var defined, key "{k}" should be in uppercase.'
+                    "All value in `env` list must follow format ENVAR=value"
                 )
     elif isinstance(value, dict):
-        if not all(k.isupper() for k in value):
-            raise InvalidArgument("All envars should be in UPPERCASE.")
+        uppercase_rgx = re.compile(r"(^[A-Z_]+\w*)")
```

Comment:
This doesn't ensure uppercase right now, only that the name starts with an uppercase character. But I agree we probably shouldn't enforce this.